teps/draft-20210126-pipeline-task-display-name.md

```diff
@@ -0,0 +1,119 @@
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+  - [Use Cases (optional)](#use-cases-optional)
+- [Requirements](#requirements)
+- [Proposal](#proposal)
+  - [Notes/Caveats (optional)](#notescaveats-optional)
+  - [Risks and Mitigations](#risks-and-mitigations)
+  - [User Experience (optional)](#user-experience-optional)
+  - [Performance (optional)](#performance-optional)
+- [Design Details](#design-details)
+- [Test Plan](#test-plan)
+- [Design Evaluation](#design-evaluation)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+- [Infrastructure Needed (optional)](#infrastructure-needed-optional)
+- [Upgrade &amp; Migration Strategy (optional)](#upgrade--migration-strategy-optional)
+- [References (optional)](#references-optional)
+<!-- /toc -->
+
+## Summary
+
+A task in a Pipeline is currently represented in the UI using a field (`name`) that is meant to be
+machine readable, not human readable. There should be the addition of a new
+optional field (`displayName`) to describe Pipeline tasks that are human readable.
+
+## Motivation
+
+Simple version, humans want to read task names formatted for human's not computers, and they want
+to see the full task name rendered in their UI.
+
+
+The longer version, the end user of a Tekton pipeline can vary including but not limited to
+application developer, to security professional, to compliance officer,
+service reliability engineer, product manager, and beyond. The farther
+you move away from the application developer the more important human
+understanding of the CI/CD process is important.
+
+Currently if a product manager or auditor was to look at a Tekton Pipeline run rendered in UI
+that would be presented with a bunch of truncated machine readable task names, no spaces, no
+capitalization, and the entire task name isn't even readable beyond 20 or so characters.
+
+Compare this to other tools in this space, Jenkins, GitLab CI, GitHub Actions, etc.
+Their renderings of their workflows are meant to be human consumable. Their task names have
+the option to be rendered in a human consumable format with use of any characters and fully
+rendered for context.
+
+### Goals
+
+* Add a `Pipeline.tasks.*.displayName` (or similar) field that allows any text and
```

Comment:
Maybe the Pipeline itself could have a displayName too ?

Reply:
Also, note that right now, catalog uses annotations for this : `tekton.dev/displayName: "git clone"`

https://github.com/tektoncd/catalog/blob/master/task/git-clone/0.2/git-clone.yaml#L10

Reply:
See also here : https://github.com/tektoncd/community/blob/master/teps/0003-tekton-catalog-organization.md#example-of-a-resource-from-catalog

Reply:
@vdemeester i did not know abou tthe `tekton.dev/displayName` i guess that would get part way there, but not all the way, since the same `Task` resource could be used multiple times in the same `Pipeline` but in different contexts for different purposes, so you would want the display name in the `Pipeline` to be different then the display name of the `Task` object.

Reply:
should i expand the goals of this to also add the display name to the `Pipeline` itself? that would be sweet too, but i don't want to make the focus of this to broad.

Reply:
add information about this to the latest push